packages/swapper/src/swappers/ChainflipSwapper/swapperApi/getTradeQuote.ts

```diff
@@ -0,0 +1,372 @@
+
+    if (fee.type === 'liquidity' && fee.asset === 'usdc.eth') return usdcAsset
+
+    if (fee.type === 'network') return usdcAsset
+  }
+
+  const getProtocolFees = (singleQuoteResponse: ChainflipBaasQuoteQuote) => {
+    const protocolFees: Record<AssetId, ProtocolFee> = {}
+
+    for (const fee of singleQuoteResponse.includedFees!) {
+      if (fee.type === 'broker') continue
+
+      const asset = getFeeAsset(fee)!
+      if (!(asset.assetId in protocolFees)) {
+        protocolFees[asset.assetId] = {
+          amountCryptoBaseUnit: '0',
+          requiresBalance: false,
+          asset,
+        }
+      }
+
+      protocolFees[asset.assetId].amountCryptoBaseUnit = (
+        BigInt(protocolFees[asset.assetId].amountCryptoBaseUnit) + BigInt(fee.amountNative!)
+      ).toString()
+    }
+
+    return protocolFees
+  }
+
+  const getQuoteRate = (sellAmountCryptoBaseUnit: string, buyAmountCryptoBaseUnit: string) => {
+    return getRate({
+      sellAmountCryptoBaseUnit,
+      buyAmountCryptoBaseUnit,
+      sellAsset,
+      buyAsset,
+    })
+  }
+
+  const getSwapSource = (swapType: string | undefined, isBoosted: boolean) => {
+    return swapType === CHAINFLIP_REGULAR_QUOTE
+      ? isBoosted
+        ? CHAINFLIP_BOOST_SWAP_SOURCE
+        : CHAINFLIP_SWAP_SOURCE
+      : isBoosted
+      ? CHAINFLIP_DCA_BOOST_SWAP_SOURCE
+      : CHAINFLIP_DCA_SWAP_SOURCE
+  }
+
+  const quotes: TradeQuote[] = []
+
+  for (const singleQuoteResponse of quoteResponse) {
+    const isStreaming = singleQuoteResponse.type === CHAINFLIP_DCA_QUOTE
+    const feeData = await getFeeData()
+
+    if (isStreaming && !deps.config.REACT_APP_FEATURE_CHAINFLIP_DCA) {
+      // DCA currently disabled - Streaming swap logic is very much tied to THOR currently and will deserve its own PR to generalize
+      // Even if we manage to get DCA swaps to execute, we wouldn't manage to properly poll with current web THOR-centric arch
+      continue
+    }
+
+    if (singleQuoteResponse.boostQuote) {
+      const boostRate = getQuoteRate(
+        singleQuoteResponse.boostQuote.ingressAmountNative!,
+        singleQuoteResponse.boostQuote.egressAmountNative!,
+      )
+
+      const boostTradeQuote: TradeQuote = {
+        id: uuid(),
+        rate: boostRate,
+        receiveAddress,
+        potentialAffiliateBps: commissionBps,
+        affiliateBps: commissionBps,
+        isStreaming,
+        slippageTolerancePercentageDecimal:
+          input.slippageTolerancePercentageDecimal ??
+          getDefaultSlippageDecimalPercentageForSwapper(SwapperName.Chainflip),
+        steps: [
+          {
+            buyAmountBeforeFeesCryptoBaseUnit: singleQuoteResponse.boostQuote.egressAmountNative!,
+            buyAmountAfterFeesCryptoBaseUnit: singleQuoteResponse.boostQuote.egressAmountNative!,
+            sellAmountIncludingProtocolFeesCryptoBaseUnit:
+              singleQuoteResponse.boostQuote.ingressAmountNative!,
```

Comment:
q: buy amount and sell amount is the same, doesn't it sound weird?

Reply:
I guess you can calculate by using egressAmountNative and substracting includedFees from it (filtering on types which make sense for shapeshift to count as fees)

An example repsonse:

```json
   "egressAsset": "usdc.arb",
    "egressAmount": 7729.312865,
    "egressAmountNative": "7729312865",
    "includedFees": [
      {
        "type": "broker",
        "asset": "usdc.eth",
        "amount": 11.624547,
        "amountNative": "11624547"
      },
      {
        "type": "ingress",
        "asset": "flip.eth",
        "amount": 5.897594859029325662,
        "amountNative": "5897594859029325662"
      },
      {
        "type": "network",
        "asset": "usdc.eth",
        "amount": 7.757456,
        "amountNative": "7757456"
      },
      {
        "type": "egress",
        "asset": "usdc.arb",
        "amount": 0.075728,
        "amountNative": "75728"
      },
      {
        "type": "liquidity",
        "asset": "flip.eth",
        "amount": 2.497051202570485337,
        "amountNative": "2497051202570485337"
      },
      {
        "type": "liquidity",
        "asset": "usdc.eth",
        "amount": 3.869036,
        "amountNative": "3869036"
      }
    ],
```

Reply:
@NeOMakinG not sure what we mean here, sellAmount is using ingress (inbound) while buyAmounts are using egress (outbounds)? 

If we mean the *after fees* bit, then that's actually correct, since those fees are paid in different assets and don't require balance (a la CoW) and are effectively a downside, as can be seen on `requiresBalance: false` in 

https://github.com/shapeshift/web/blob/5da50735d021e0383b2168e0233bf60753116271/packages/swapper/src/swappers/ChainflipSwapper/swapperApi/getTradeQuote.ts#L219
<img width="488" alt="image" src="https://github.com/user-attachments/assets/58d4a289-c823-44f0-a774-62a39703ff1f">